src/ui/components/NetworkMonitor/index.tsx

```diff
@@ -89,49 +89,51 @@ export const NetworkMonitor = ({
 
   return (
     <Table events={events} requests={requests} types={types}>
-      {({ table, data }: { table: any; data: RequestSummary[] }) => (
-        <div className="flex h-full min-h-0 flex-col" ref={container}>
-          <FilterBar types={types} toggleType={toggleType} table={table} />
-          <SplitBox
-            className="min-h-0 border-t border-splitter"
-            initialSize="350px"
-            minSize={selectedRequestId ? "30%" : "100%"}
-            maxSize={selectedRequestId ? "70%" : "100%"}
-            startPanel={
-              <RequestTable
-                table={table}
-                data={data}
-                currentTime={currentTime}
-                onRowSelect={row => {
-                  trackEvent("net_monitor.select_request_row");
-                  dispatch(fetchFrames(row.point));
+      {({ table, data }: { table: any; data: RequestSummary[] }) => {
+        let selectedRequest;
+        if (selectedRequestId) {
+          selectedRequest = data.find(request => request.id === selectedRequestId);
```

Comment:
If it's unexpected that we'd ever _not_ find a match here, I could add a `console.warn` log or something.